client/api.c

```diff
@@ -733,6 +742,112 @@ TDNFOpenHandle(
     goto cleanup;
 }
 
+uint32_t
+TDNFAddCmdLinePackages(
+    PTDNF pTdnf
+)
+{
+    uint32_t dwError = 0;
+    PTDNF_CMD_ARGS pCmdArgs = NULL;
+    PSolvSack pSack;
+    int nIsFile;
+    int nIsRemote;
+    int nCmdIndex;
+    char *pszPkgName;
+    char *pszCopyOfPkgName = NULL;
+    char* pszRPMPath = NULL;
+    char cwd[PATH_MAX];
+    Id id;
+
+    if(!pTdnf)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pCmdArgs = pTdnf->pArgs;
+    pSack = pTdnf->pSack;
+
+    for(nCmdIndex = 1; nCmdIndex < pCmdArgs->nCmdCount; ++nCmdIndex)
+    {
+        pszPkgName = pCmdArgs->ppszCmds[nCmdIndex];
+
+        dwError = TDNFIsFileOrSymlink(pszPkgName, &nIsFile);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        if (nIsFile)
+        {
+            if (pszPkgName[0] != '/')
+            {
+                getcwd(cwd, sizeof(cwd));
+
+                dwError = TDNFAllocateStringPrintf(
+                              &pszRPMPath,
+                              "%s/%s",
+                              cwd,
+                              pszPkgName);
+                BAIL_ON_TDNF_ERROR(dwError);
+            }
+            else
+            {
+                dwError = TDNFAllocateString(pszPkgName, &pszRPMPath);
+                BAIL_ON_TDNF_ERROR(dwError);
+            }
+        }
+        else
+        {
+             dwError = TDNFUriIsRemote(pszPkgName, &nIsRemote);
```

Comment:
In this case the pattern is actually useful, because we have 3 (interesting) states returning from `TDNFUriIsRremote()`: remote URL, local URL (file://), or not a URL at all; and not just a boolean.

But besides that, I would agree with you if this pattern wasn't so strongly used already throughout all tdnf code. With new code I would agree that this is odd, but I think breaking this pattern in tdnf would make it more confusing.